### PR TITLE
Feature/lp140 input chips

### PR DESCRIPTION
**Pull request by juanislas77**
Make Input Chip
https://devaptivist.atlassian.net/browse/LP-140

Create a generic chip
image

### CODE REVIEW

**Review by carlosph816 — APPROVED** [review]
Well done!

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    onClick: () -> Unit
+) {
+    val sizes = LeasePertTheme.sizes
+    Box(
+        modifier = modifier
+            .padding(end = sizes.minorSmallSize, bottom = sizes.minorSmallSize)
+            .clickable { onClick }
+    ) {
+        Box(
+            modifier = modifier
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: Do you think if is there any better way to apply the Box to use just one of them?

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+fun LpGenericChip(
+    modifier: Modifier = Modifier,
+    label: String,
+    icon: ImageVector? = null,
+    onClick: () -> Unit
+) {
+    val sizes = LeasePertTheme.sizes
+    Box(
+        modifier = modifier
+            .padding(end = sizes.minorSmallSize, bottom = sizes.minorSmallSize)
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: I think this padding is unnecessary like the parameter modifier is who defines all the outside paddings of the component

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package com.iteneum.designsystem.components
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt:1]: Your button doesn't look the same as the design

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        modifier = modifier
+            .padding(end = sizes.minorSmallSize, bottom = sizes.minorSmallSize)
+            .clickable { onClick }
+    ) {
+        Box(
+            modifier = modifier
+                .clip(RoundedCornerShape(sizes.midSmallSize))
+                .background(MaterialTheme.colorScheme.primary)
+                .padding(
+                    start = sizes.smallerSize,
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: You can change all paddings to the same size as it not looks like the design

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                    top = sizes.minorSmallSize,
+                    end = sizes.smallerSize,
+                    bottom = sizes.minorSmallSize
+                )
+                .defaultMinSize(minWidth = sizes.extraSize36, minHeight = sizes.regularSize),
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: I think you dont need the default min size, or what do you think?

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                icon?.let {
+                    Icon(
+                        imageVector = icon,
+                        contentDescription = stringResource(R.string.icon_description),
+                        modifier = Modifier.size(ButtonDefaults.IconSize),
+                        tint = MaterialTheme.colorScheme.onPrimary
+                    )
+                    Spacer(modifier = Modifier.size(ButtonDefaults.IconSpacing))
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: Another thing that you could do is remove the spacer and add padding to the icon

File: core/designsystem/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -4,4 +4,5 @@
     <string name="OTFPasswordMessageError">At least 6 characters</string>
     <string name="cd_Icon">icon image</string>
     <string name="DTFDropdownDescription">Dropdown Arrow</string>
+    <string name="icon_description">an icon</string>
```
Comment (melissaquezada94) [inline core/designsystem/src/main/res/values/strings.xml:9]: Please check the nomenclature for the strings

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import com.iteneum.designsystem.R
+import com.iteneum.designsystem.theme.LeasePertTheme
+
+@Composable
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt:31]: Please add the documentation to this function

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt (hunk 6 above)
Comment (juanislas77) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: Done!

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt (hunk 8 above)
Comment (juanislas77) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt:31]: Done!

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt (hunk 2 above)
Comment (juanislas77) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: Done!

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt (hunk 1 above)
Comment (juanislas77) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: Done!

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt (hunk 5 above)
Comment (juanislas77) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: Done!

**Comment by juanislas77** [comment]
image

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt (hunk 4 above)
Comment (juanislas77) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/Chips.kt]: Done! 👍🏽

**Review by melissaquezada94 — APPROVED** [review]
well done

**Comment by juanislas77** [comment]
Ready to merge

**Pull request merged**